addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200ErrorService.java

```diff
@@ -0,0 +1,146 @@
+     */
+    void removeAllErrors() {
+        synchronized (errorMap) {
+            if (errorMap != null) {
+                errorMap.clear();
+            }
+        }
+    }
+
+    /**
+     * This function updates the errors
+     *
+     */
+    void updateErrors(JsonObject nodeRoot) {
+        synchronized (errorMap) {
+            /* Update the list of errors */
+            try {
+                removeAllErrors();
+                JsonArray sPoints = nodeRoot.get("values").getAsJsonArray();
+                for (int i = 0; i < sPoints.size(); i++) {
+                    JsonObject subJSON = sPoints.get(i).getAsJsonObject();
+                    HashMap<String, String> valMap = new HashMap<String, String>();
+                    Set<Map.Entry<String, JsonElement>> oMap = subJSON.entrySet();
+                    oMap.forEach(item -> {
+                        logger.debug("Set: {} val: {}", item.getKey(), item.getValue().getAsString());
+                        valMap.put(item.getKey(), item.getValue().getAsString());
+                    });
+                    errorMap.add(valMap);
+                }
+            } catch (Exception e) {
+                logger.error("Error in parsing of the errorlist: {}", e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * This function returns the number of errors
+     *
+     */
+    Integer getNbrErrors() {
+        synchronized (errorMap) {
+            return errorMap.size();
+        }
+    }
+
+    /**
+     * This function sets the actual errors
+     *
+     */
+    void setActiveError(Integer error) {
+        Integer actError;
+        if (error < 1) {
+            actError = 1;
+        } else if (error > getNbrErrors()) {
+            actError = getNbrErrors();
+        } else {
+            actError = error;
+        }
+        synchronized (activeError) {
+            activeError = actError;
+        }
+    }
+
+    /**
+     * This function returns the selected error
+     *
+     */
+    Integer getActiveError() {
+        synchronized (activeError) {
+            return activeError;
+        }
+    }
+
+    /**
+     * This function returns a error string with all parameters
+     *
+     */
+    String getErrorString() {
+        String value = "";
+        synchronized (errorMap) {
+            Integer actN = getActiveError();
+            if (errorMap.size() < actN || errorMap.size() == 0) {
+                return null;
+            }
+            /* is the time value existing ("t") the use it on the begin */
+            if (errorMap.get(actN - 1).containsKey("t")) {
+                value = errorMap.get(actN - 1).get("t");
+                for (String para : errorMap.get(actN - 1).keySet()) {
+                    if (!"t".equals(para)) {
+                        value += " " + para + ":" + errorMap.get(actN - 1).get(para);
+                    }
+                }
+            } else {
+                for (String para : errorMap.get(actN - 1).keySet()) {
```

Comment:
Check for other usages of `para` and give meaningful names, also note you should better use a entrySet here because you want to use the keys and the values. Using the entrySet is way more efficient. 

Please check for other occurrences as well, I already see one other at 133 but there might be more.